packages/core/src/components/DataTable/index.tsx

```diff
@@ -111,6 +111,14 @@ export class DataTable extends React.Component<DataTableProps & WithStylesProps,
     outline: 'none',
   });
 
+  static getDerivedStateFromProps(props: DataTableProps, state: State) {
+    if (indexDataList(props.data!) != state.sortedDataList) {
```

Comment:
I think here the condition will be always `true`..  And we don't want to call `indexDataList(props.data)` twice (here once, the line underneath the second time).

Reply:
If the performance is major consideration, we could memoize the result using for example https://github.com/caiogondim/fast-memoize.js/

Reply:
In storybook I ran a test with ~2k rows and had no issues, but it's definitely a consideration and something I'm a little worried about.

If a user/developer does start hitting this, I think the biggest improvement would be optional data version numbers:
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#preferred-solutions

Otherwise even with memoization I think we're having to compare/hash the entire data input which is faster, but still similar to the current method.

Reply:
What I meant here is that `indexDataList(props.data)` will be always not equal to `state.sortedDataList`, since they are two different array.